src/test/ui/nll/relate_tys/hr-fn-aau-eq-abu.rs

```diff
@@ -0,0 +1,38 @@
+// Copyright 2017 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+// Test an interesting corner case: a fn that takes two arguments that
+// are references with the same lifetime is in fact equivalent to a fn
+// that takes two references with distinct lifetimes. This is true
```

Comment:
... you should probably revise this comment, or at least elaborate a bit that while in principle such fn's are equivalent, our checks are currently distinguishing them?

Reply:
(or are you figuring that the FIXME down in the body suffices for that purpose? I just find the difference between the header vs the actual behavior jarring.)

Reply:
Yeah, will fix.